master/internal/task/allocation.go

```diff
@@ -444,6 +448,11 @@ func (a *Allocation) TaskContainerStateChanged(
 			prom.RemoveAllocationReservation(a.reservations[cID].Summary())
 		}
 	}
+
+	err := a.db.UpdateAllocationState(a.req.AllocationID, a.state, a.logBasedReadinessPassed)
```

Comment:
i think this would can just be a `ctx.Log().Error()` and the one above it should `a.Error()` and return. wdyt?